test/bin/scenario.sh

```diff
@@ -1128,12 +1135,45 @@ run_gingko_tests() {
     export KUBECONFIG="${kubeconfig}"
     record_junit "${vmname}" "setup_kubeconfig" "OK"
 
+    # Create case selection file
+    local case_selected="${test_results_dir}/case_selected"
+
+    # Get all MicroShift tests using dry-run
+    "${GINKGO_TEST_BINARY}" run all --dry-run | grep -E "MicroShift" > "${case_selected}"
+
+    # Apply filter if provided
+    if [[ -n "${filter}" ]]; then
+        echo "Applying filter: ${filter}"
+        local temp_file="${case_selected}.tmp"
+
+        if [[ "${filter}" == ~* ]]; then
```

Comment:
Instead of this pseudo parsing of filter expression, can we provide filters in grep -E format? 
This should allow us to simplify the logic.

Reply:
I would want to support both inclusion and exclusion patterns in the same filter list and using it this way i feel it would be much simpler and intuitive and matches the current QE pipelines.